scripts: fix zstd companion library usage

Lapshin opened the pull request:
Proper fix that was introduced in https://github.com/crosstool-ng/crosstool-ng/pull/2144

- added proper AR for zstd build to fix macos builds
- zstd was never installed properly because of the absence of a pkg-config file
- binutils option `--with-zstd=` does not process paths (see https://github.com/bminor/binutils-gdb/blob/200164d46732f6a2fb9d81aa9650c006db4148e6/config/zstd.m4)

Lapshin committed:
scripts: fix zstd companion library usage

Signed-off-by: Alexey Lapshin <alexey.lapshin@espressif.com>
[cp: Add to gdb-cross, remove from gdb-native]
Signed-off-by: Chris Packham <judge.packham@gmail.com>

cpackham commented:
@Lapshin looks like there was a build failure related to this
[powerpc64-unknown-linux-musl.ubuntu-22.04.log.zip](https://github.com/user-attachments/files/24919958/powerpc64-unknown-linux-musl.ubuntu-22.04.log.zip)

cpackham committed:
musl: Add mirror

musl seems to be getting hit with the hug of death from AI scrapers.
Like we did in commit 6cb4b26c ("uClibc-ng: Add mirror") piggy-back off
the buildroot package mirror for musl.

Signed-off-by: Chris Packham <judge.packham@gmail.com>
